utilities/test_suite/rpp_test_suite_common.h

```diff
@@ -529,13 +807,85 @@ inline void remove_substring(string &str, string &pattern)
+        int matched_idx = 0;
+        int refOutputHstride = refOutputWidth * dstDescPtr->c;
+
+        for(int i = 0; i < height; i++)
+        {
+            rowTemp = outputTemp + i * dstDescPtr->strides.hStride;
+            rowTempRef = outputTempRef + i * refOutputHstride;
+            for(int j = 0; j < width; j++)
+            {
+                outVal = rowTemp + j;
+                outRefVal = rowTempRef + j ;
+                int diff = abs(*outVal - *outRefVal);
+                if(diff <= CUTOFF)
+                    matched_idx++;
+            }
+        }
+        if(matched_idx == (height * width) && matched_idx !=0)
+            fileMatch++;
+    }
+}
+
+// compares the output of PLN3-PLN3 and PLN1-PLN1 variants
+void compare_pln(Rpp8u* output, Rpp8u* refOutput, RpptDescPtr dstDescPtr, RpptImagePatch *dstImgSizes, int refOutputHeight, int refOutputWidth, int refOutputSize, int &fileMatch)
+{
+    Rpp8u *rowTemp, *rowTempRef, *outVal, *outRefVal, *outputTemp, *outputTempRef, *outputTempChn, *outputTempRefChn;
+    for(int imageCnt = 0; imageCnt < dstDescPtr->n; imageCnt++)
+    {
+        outputTemp = output + imageCnt * dstDescPtr->strides.nStride;
+        outputTempRef = refOutput + imageCnt * refOutputSize;
+        int height = dstImgSizes[imageCnt].height;
+        int width = dstImgSizes[imageCnt].width;
+        int matched_idx = 0;
+        int refOutputHstride = refOutputWidth;
+        int refOutputCstride = refOutputHeight * refOutputWidth;
+
+        for(int c = 0; c < dstDescPtr->c; c++)
+        {
+            outputTempChn = outputTemp + c * dstDescPtr->strides.cStride;
+            outputTempRefChn = outputTempRef + c * refOutputCstride;
+            for(int i = 0; i < height; i++)
+            {
+                rowTemp = outputTempChn + i * dstDescPtr->strides.hStride;
+                rowTempRef = outputTempRefChn + i * refOutputHstride;
+                for(int j = 0; j < width; j++)
+                {
+                    outVal = rowTemp + j;
+                    outRefVal = rowTempRef + j ;
+                    int diff = abs(*outVal - *outRefVal);
+                    if(diff <= CUTOFF)
+                        matched_idx++;
+                }
+            }
+        }
+        if(matched_idx == (height * width * dstDescPtr->c) && matched_idx !=0)
+            fileMatch++;
+    }
+}
+
 template <typename T>
-inline void compare_output(T* output, string funcName, RpptDescPtr srcDescPtr, RpptDescPtr dstDescPtr, RpptImagePatch *dstImgSizes, int noOfImages, string interpolationTypeName, int testCase, string dst)
+inline void compare_output(T* output, string funcName, RpptDescPtr srcDescPtr, RpptDescPtr dstDescPtr, RpptImagePatch *dstImgSizes, int noOfImages, string interpolationTypeName, string noiseTypeName, int testCase, string dst)
 {
     string func = funcName;
     string refPath = get_current_dir_name();
     string pattern = "/build";
     string refFile = "";
+    int refOutputWidth = 152;
```

Comment:
Also mention logic for 152 ((150 / 8) * 8 + 8 = 152) in the comment

Reply:
// obtain next multiple of 8 after 150 as ((150 / 8) * 8 + 8 = 152)

Reply:
Done